### PR TITLE
Show thank you page after a donation has made

### PR DESCRIPTION
**Pull request by orlando**
**What:** Show thank you page after a donation has made.

https://www.loom.com/share/d39aaf400d71452c864d735cef79ea21

**Why:** Related to https://app.asana.com/0/1168997577035609/1185366458136044

**How:**

- [x] Add thank you page
- [x] Redirect to thank you page after donation is made
- [x] Add name, email and address required fields to checkout form
- [x] Send thank you email after donation with donation information
- [x] Add endpoint for users to review their receipt

### CODE REVIEW

**Comment by duranmla** [comment]
I think the pull request is growing too much. Also, I don't mean to force opinions over your workflow but having (wip) commits sounds fundamentally odd, for instance:

```
dev(emails): add thank you email template (wip)
dev(emails): add email layout, add thank you email template (wip)
```

Means that you two times add the email template? why not add a section of the template completely and then another section such as:

```
dev(emails): add thank you email template base layout
dev(emails): add thank you email template header
dev(emails): add thank you email template body
```

I think that worth to mention that the obscure the pull request the harder is to get a good review _(which often is your case due to work with context that not all the team handles - we very much appreciate it -)_ With that being said I will recommend you to:

- Split your work in batches: will make you happier delivering value everyday and have transparent and easy to review PR
- Take care of you commit messages: some people use it in order to understand your intend with the code 

In this case, I can see the work in the following PRs
**PR#1**: Update all dependencies, fix test, cleanup gemfiles
**PR#2**: Dry up warnings and errors from the code _(isolate this will create a point of reference for further developers to see the mistakes made before)_
**PR#3**: Add the logic so send the thank you email and redirect to show thank you page _(arguable something to split)_
**PR#4**: Email template we are gonna use _(if that is going to be defined here)_

Of course, the split is something that is totally up to you and your way of work I am just giving to you the perspective from a viewer feel free to act as you feel makes more sense for you.

**Comment by orlando** [comment]
@duranmla Yeah, I agree, the first bunch of commits should have been split to another PR since are mostly are fixing DX. About the wip commits, I remove these with a rebase before merging a PR, so don't worry about it.

I'm going to clean up this PR and make it ready for review and open new ones for the other tasks.

**Comment by orlando** [comment]
@duranmla @arathjz this is ready for review, will continue with the remaining tasks on another PR

File: app/javascript/packs/application.js
Comment (duranmla) [inline app/javascript/packs/application.js:23]: Pretty sure you have a reason for this but... Why you can not attach the click on the component itself and instead you have to wait for "DOMContentLoaded"?

File: app/views/shared/_stripe_scripts.html.erb
Comment (duranmla) [inline app/views/shared/_stripe_scripts.html.erb]: I guess this shouldn't be here

File: app/views/static_pages/home.html.erb
Comment (duranmla) [inline app/views/static_pages/home.html.erb]: That wording may need a ticket in order to think about it:

> Folks who donate $100 on a one-time basis will receive a free gift from our store. People who pay dues at the rate of $100 per month or more will receive a thank you call from us.

For sure is the truth but highlight in a very explicit way the difference and feels like undervaluing folks with dues of rate that $100 per month.

Besides, how come that someone with subscription of 100$ doesn't get a gift and a person with ONE time donation does?

An alternative will be something like:

> Receive a personal call from us to thank you about you support by joining into the movement with 100$ subscription or get a special gift from our store after one time donation of 100$ or more.

File: app/assets/stylesheets/_alerts.scss
Comment (duranmla) [inline app/assets/stylesheets/_alerts.scss:6]: Avoid `px` usage and always go with `rem`. While modern browsers can scale `px` _(thus the premise of respect user preference is getting old)_ it is still a good practice to unify the units we use to define styles to avoid questions and confusion with new developers.

**Review by duranmla — APPROVED** [review]
Looks good, couple comments but nothing blocker. Thanks!!

**Comment by duranmla** [comment]
@orlando kind reminder to remove the "WIP" from the title.
_Also, maybe next time you can use draft https://github.blog/2019-02-14-introducing-draft-pull-requests/_

File: app/views/static_pages/home.html.erb
Comment (orlando) [reply app/views/static_pages/home.html.erb]: I won't change copy on this PR, this will come on the redesign. But I do believe we need to remove this since is not valid. we don't send any gifts or anything when people donate. I'll check if I can remove this copy

File: app/assets/stylesheets/_alerts.scss
Comment (orlando) [reply app/assets/stylesheets/_alerts.scss:6]: for borders, I don't think it makes sense to use rems. Borders width shouldn't change with screen size and are meant to be fixed. Also, browsers have problems rendering sub-pixel borders https://chenhuijing.com/blog/about-subpixel-rendering-in-browsers.

I do agree to have these for margin and padding though.

File: app/javascript/packs/application.js
Comment (orlando) [reply app/javascript/packs/application.js:23]: I'm not sure if this is an actual issue, but for correctness it should be done this way. If the element isn't present on the first load (this comes from the server so it should be there before window.onload triggers) it will fail to bind the click event.

**Comment by orlando** [comment]
@duranmla removed the gift copy from the plan's description and addressed the other comments 
image. Will merge once it's green

**Pull request merged**
